### ISSUE TITLE
Add a `shouldReturn` or `didReturn` hook

### ISSUE THREAD

**Issue by irace** [stripe/stripe-ios#314]
When part of a larger form, it’s useful to be able to to move directly to the next field once the credit card field is done being populated.

**Comment by jflinter** [comment]
@irace - agreed, but there are a couple of problems with this.

- The numeric keyboard doesn't have a return key. Inexplicably.
- You're not always certain a user is done typing. In the annoying case of 3-digit amex CVCs, the user may or may not have a 4th digit to type after typing 3 digits.

I've considered adding a method to `STPPaymentCardTextField` like `automaticallyShowInputAccessoryWithTarget:action:` that'll wire up a `UIToolbar` with a done item as the textField's `inputAccessoryView`, and auto-enable the done item depending on its internal state. Thoughts?

**Comment by irace** [comment]
Not a bad idea, though also probably should be a pretty low priority.

**Comment by jflinter** [comment]
Oh, don't worry, it is.
